app/controllers/auth/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Auth::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+
+  def self.provides_callback_for(provider)
+    _provider = provider.to_s.chomp "_oauth2"
+    class_eval %Q{
```

Comment:
Yes, we could indeed do it without the scary `class_eval`. It's just that the callback/process is pretty much the same for any provider you would decide to add (or not) in the future.